### ISSUE TITLE
Milestone 4 - Address Peer Review and TA Feedback

### ISSUE THREAD

**Issue by jadeeechen** [jasjotp/relationship-quality-prediction#46]
### 3.1 Address feedback from TAs

After you make changes based on feedback from your TAs, please raise a regrade requests on Gradescope for the corresponding question, **provide links to the PRs, commit history to demonstrate the changes you made**. After the TA confirm your changes are made correctly, you will be awarded **50% of the points you lost for that question**.

## ⚠️ Important:

-   Deadline for Milestone 1 + 2 regrade requests: **Saturday, December 13, 2025 at 6PM.**
-   Deadline for Milestone 3 regrade requests: **Wednesday, December 17th, 2025 at 11:59 AM**.
-   Your regrade request should be clear to read, easy to track your changes, otherwise we will not be able to give points back.

### 3.2 Address feedback from peer review

After you receive feedback from your peers, each group member should at least address one feedback item from the peer review. If you are in a group of 3, you should address at least 3 feedback items from the peer review. If you are in a group of 4, you should address at least 4 feedback items from the peer review.

To help us easily, and correctly, assess changes you made based on peer feedback, please create a file called `CHANGELOG.md` that lives in the root of your repository. In this file, please describe any improvements you made to the project based on peer feedback (if peer feedback is the same as feedback from your TAs, that is ok to include them in CHANGELOG too), and point to evidence of these improvements.

You can point us to evidence of addressing it by creating an repository issue, and linking to another issue and pull request.
providing URLs, links to PRs, links to commits, reference specific lines of code, etc can all be used to track how a feature/bug was fixed.
Be sure to add some narration when sharing these URLs so that it is easy for us to identify which changes to your work addressed which pieces of feedback.

See this page for an example of a changelog and how you can reference fixes: <https://keepachangelog.com/>

**Comment by jadeeechen** [comment]
Below is a compiled list of tasks from TA feedback:

**Milestone 1**

- `README.md`: Included dependencies but missed to include version requirement for dependencies
- `CODE_OF_CONDUCT.md`: Missing an email contact about who to contact if someone breaks the code of conduct. 
- `<analysis>.ipynb` or `<analysis>.Rmd`: Document does not have a descriptive name (it should not be named analysis.ipynb for example).
- `LICENSE.md`: No Creative Commons license (for project report) was specified. LICENSE file did not contain names of all group members
- Abstract/Summary: Does not clearly report the major findings. Does not discuss importance and limitations of findings.
- Methods:
  1.  You have broken the golden rule of machine learning. You did EDA on the entire dataset before splitting it to train and test. We should always split the dataset to train and test, then do EDA on the train set only. this is because we should not peek at the "final exam" (i.e., test set) in the beginning. 
  2. There are also coding issues like, you modified the data type for original dataframe X after you split the data into train and test, not before: 
    ```# split data into train and test split
    X_train, X_test, y_train, y_test = train_test_split(
        X, y, test_size = 0.2, random_state = 123
    )
    
    # simple data cleaning - change subject age and children from floats to intgers 
    X['subject_age'] = X['subject_age'].astype(int)
    X['children'] = X['children'].astype(int)
    ```
  3. You did not mention what metric you used, i know it is accuracy by default, and you used ROC. but this needs to be written clearly in the methods section, which metrics you used, why you used it. 
  4. Your dataset is unbalanced. please describe how you handles this imbalance in your methods.
  5. Please also clearly add Methods section and Results section title
- Results:
  1. The font size is too small for plots, especially the bar plots.
  2. There are variable names displayed on the plot, please use phrases separated by whitespace, not underscore. We use underscore in variable naming, and white space to seperate words when we are writing papers and report. This plot is showing your result to general readers.
- Discussion:
  1. Some key assumptions and limitations are missing. For example, one limitation is this study only tested 1 model, but we could have tested more model.
  2. There is also space for hyperparameter optimization to be done here to improve the model performance just for logistic regression. I would not jump to the conclusion that the low performance of this one model means that these features we are using or not good. 
- References: There are references at the end of the report, but they are not referred to in the narrative.
- Grammar: Grammar mistakes and spelling errors (e.g., "...a dataset on based on ..."; "relationshup")
- Reproducibility: I was able to recreate the conda environment correctly, but I can not run the jupyter notebook analysis in the jupyter lab, because your conda environment is missing packages like ipykernel, to help us use your conda environment as a jupyter kernel. 
- GitHub issues for communication: Please use GitHub issues to document your workflow, communications, distribution of work.
    
**Milestone 2**

- `docker-compose.yml`:
  1. You should specify a version tag for the image to ensure reproducible builds.
  2. You should mount only the necessary directories instead of the entire project to make running the analysis cleaner and easier.
- Data Validation:
  1. Missing check: "No anomalous correlations between target/response variable and features/explanatory variables" 
  2. "Target/response variable follows expected distribution" – You should define and justify thresholds for expected distribution carefully; for instance, the validation checks that excellent <= 0.6, good <= 0.4, etc., which are arbitrary and may not reflect realistic distribution expectations.
- Reproducibility: You are expected to update README.md to include step-by-step instructions on how to use the project.

**Milestone 3**

NO FEEDBACK YET

**Milestone 4**

NO FEEDBACK YET
